drivers/bridge/setup_ipv6.go

```diff
@@ -117,3 +120,31 @@ func setupIPv6Forwarding(config *networkConfiguration, i *bridgeInterface) error
 
 	return nil
 }
+
+func setupNDPProxying(config *networkConfiguration, i *bridgeInterface) error {
+	// Get current NDP default proxying setup
+	ndpProxyDataDefault, err := ioutil.ReadFile(ndpProxyConfDefault)
+	if err != nil {
+		return fmt.Errorf("Cannot read NDP default proxying setup: %v", err)
+	}
+	// Enable NDP default proxying only if it is not already enabled
+	if ndpProxyDataDefault[0] != '1' {
+		if err := ioutil.WriteFile(ndpProxyConfDefault, []byte{'1', '\n'}, ndpProxyConfPerm); err != nil {
+			logrus.Warnf("Unable to enable NDP default proxying: %v", err)
+		}
+	}
+
+	// Get current NDP all proxying setup
+	ndpProxyDataAll, err := ioutil.ReadFile(ndpProxyConfAll)
+	if err != nil {
+		return fmt.Errorf("Cannot read NDP all proxying setup: %v", err)
+	}
+	// Enable NDP all proxying only if it is not already enabled
+	if ndpProxyDataAll[0] != '1' {
```

Comment:
No, the first loop you pointed out is only to detect if the to-be-created network's configuration conflicts with any existing network. And it ends at https://github.com/mostolog/libnetwork/blob/patch-1/drivers/bridge/bridge.go#L615

Then, the second line you linked
```
enableNDPProxying := config.NDPProxyInterface != "" && config.AddressIPv6 != nil
```
it's where the driver detects whether it does or does not need to program the ndp proxying for this network. This is just fine and does not need any changes.

The only one I was asking for is to see if it is enough for `setupNDPProxying()` to only enable proxying for the interface passed in the driver options, instead of enabling it for all the interfaces in the host.

The setup function has access to the ndp interface name because it is passed the network configuration:
`func setupNDPProxying(config *networkConfiguration, i *bridgeInterface) error {` has acces to 

So I was asking that instead of etting `ndpProxyConfAll`, we should set 
```
ndpProxyConfPath := fmt.Sprintf("/proc/sys/net/ipv6/conf/%s/proxy_ndp", config.NDPProxyInterface)
if err := ioutil.WriteFile(ndpProxyConfPath, []byte{'1', '\n'}, ndpProxyConfPerm); err != nil {
		logrus.Warnf("Unable to enable NDP proxying for interface %s: %v", config.NDPProxyInterface,err)
	}
```